src/component/project/Project/Project.tsx

```diff
@@ -8,88 +8,65 @@ import { Tab, Tabs } from '@mui/material';
-    const activeTab = useOptionalPathParam('activeTab');
     const params = useQueryParams();
     const { project, error, loading, refetch } = useProject(projectId);
     const ref = useLoading(loading);
     const { setToastData } = useToast();
     const { classes: styles } = useStyles();
     const navigate = useNavigate();
+    const { pathname } = useLocation();
     const { isOss } = useUiConfig();
-
     const basePath = `/projects/${projectId}`;
     const projectName = project?.name || projectId;
-    const tabData = [
-        {
-            title: 'Overview',
-            component: (
-                <ProjectOverview
-                    projectId={projectId}
-                    projectName={projectName}
-                />
-            ),
-            path: basePath,
-            name: 'overview',
-        },
-        {
-            title: 'Health',
-            component: (
-                <ProjectHealth
-                    projectId={projectId}
-                    projectName={projectName}
-                />
-            ),
-            path: `${basePath}/health`,
-            name: 'health',
-        },
-        {
-            title: 'Access',
-            component: <ProjectAccess projectName={projectName} />,
-            path: `${basePath}/access`,
-            name: 'access',
-        },
-        {
-            title: 'Environments',
-            component: (
-                <ProjectEnvironment
-                    projectId={projectId}
-                    projectName={projectName}
-                />
-            ),
-            path: `${basePath}/environments`,
-            name: 'environments',
-        },
-        {
-            title: 'Archive',
-            component: (
-                <ProjectFeaturesArchive
-                    projectId={projectId}
-                    projectName={projectName}
-                />
-            ),
-            path: `${basePath}/archive`,
-            name: 'archive',
-        },
-    ];
 
-    const activeTabIdx = activeTab
-        ? tabData.findIndex(tab => tab.name === activeTab)
-        : 0;
+    const tabs = useMemo(
+        () => [
+            {
+                title: 'Overview',
+                path: basePath,
+                name: 'overview',
+            },
+            {
+                title: 'Health',
+                path: `${basePath}/health`,
+                name: 'health',
+            },
+            {
+                title: 'Access',
+                path: `${basePath}/access`,
+                name: 'access',
+            },
+            {
+                title: 'Environments',
+                path: `${basePath}/environments`,
+                name: 'environments',
+            },
+            {
+                title: 'Archive',
+                path: `${basePath}/archive`,
+                name: 'archive',
+            },
+        ],
+        [basePath]
+    );
+
+    const activeTab = useMemo(() => {
```

Comment:
I'm not sure, but I think perhaps memoization here is going to cost more than iterating over a list of 5 items.

Reply:
Yeah you're probably right. Feel free to remove. :)